### PR TITLE
Add model field mapping

### LINKED ISSUE #198: Enable mapping a GraphQL field to a different model field
### Expected Behaviour
It'd be nice to be able to map field X in a graphql schema to a different object field via a mapping directive in `gqlgen.yaml`.

### Sample gqlgen.yaml (from https://gqlgen.com/config/)
```yaml
schema: schema.graphql

# Let gqlgen know where to put the generated server
exec:
  filename: graph/generated/generated.go
  package: generated

# Let gqlgen know where to the generated models (if any)
model:
  filename: models/generated.go
  package: models

# Tell gqlgen about any existing models you want to reuse for
# graphql. These normally come from the db or a remote api.
models:
  User:
    model: github.com/my/app/models.User
  Todo:
    model: github.com/my/app/models.Todo
    fields:
      id:
        modelField: IDString # maps to field IDString in models.Todo
```

### PR DESCRIPTION
**Pull request by vvakame**
closes #198

### CODE REVIEW

**Comment by vvakame** [comment]
👀 next one...
https://github.com/vektah/gqlgen/tree/feat-lintfree
https://github.com/vektah/gqlgen/commit/f7f6f9166ab71b67713276597347d429b4691398

File: codegen/config.go [hunk 1]
```diff
@@ -78,7 +78,8 @@ type TypeMapEntry struct {
 }
 
 type TypeMapField struct {
-	Resolver bool `yaml:"resolver"`
+	Resolver  bool   `yaml:"resolver"`
+	GoVarName string `yaml:"govarname"`
```
[the remark this answers is not in the document]
Comment (vektah) [reply codegen/config.go]: :thinking: 

What about binding to methods?

eg
```yaml
  Todo:
    model: github.com/my/app/models.Todo
    fields:
      id:
        modelField: IDString 
```

to bind to 
```go
func (t *Todo) IdString() string {
    return ...
}
```

what about `fieldName`?

[the remark this answers is not in the document]
Comment (vvakame) [reply codegen/config.go]: @vektah plz take a look again.
1. rename to `fieldName`
2. support binding to method & field both

**Comment by vvakame** [comment]
appveyor is little bit flaky...? 🤔

**Comment by vektah** [comment]
Yeah, it doesn't deal with test concurrency well, maybe their machines are overloaded.

I've been meaning to take a look, but there are bigger fish to fry.

File: example/config/server/server.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"runtime/debug"
+
+	todo "github.com/vektah/gqlgen/example/config"
```
Comment (vektah) [inline example/config/server/server.go]: aliased todo?

File: example/config/server/server.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/vektah/gqlgen/handler"
+)
+
+func main() {
+	http.Handle("/", handler.Playground("Todo", "/query"))
+	http.Handle("/query", handler.GraphQL(
+		todo.NewExecutableSchema(todo.New()),
+		handler.RecoverFunc(func(ctx context.Context, err interface{}) error {
```
Comment (vektah) [inline example/config/server/server.go]: this was copied from todo, might as well drop the error stuff?

File: codegen/config.go [hunk 4]
```diff
@@ -78,7 +78,8 @@ type TypeMapEntry struct {
 }
 
 type TypeMapField struct {
-	Resolver bool `yaml:"resolver"`
+	Resolver  bool   `yaml:"resolver"`
+	ModelName string `yaml:"modelField"`
```
Comment (vektah) [inline codegen/config.go]: fieldName?

File: example/config/server/server.go (hunk 2 above)
Comment (vvakame) [reply example/config/server/server.go]: ya, I copied from example/todo . I thought that `config` was a bit confusing

File: codegen/config.go (hunk 4 above)
Comment (vvakame) [reply codegen/config.go]: 🙀

**Pull request merged**